### DISCUSSION TITLE
Auto Update MAUI App

### DISCUSSION THREAD

**Discussion by RNavaneethKumar** [dotnet/maui#8922] [General]
I have a MAUI app that is under development and is being tested on Windows, IOS and Android. As the development is happening in parallel, there is a need to connect the devices to the development machine and deploy the app every single time. As we are in the development phase, is there any way to auto update the application during start without the app store or windows store approach? Please advice and suggest.

**Comment by anpin** [comment]
have you seen https://github.com/dotnet/xharness ?

**Comment by RNavaneethKumar** [comment]
Sorry if I did not explain the issue more clearly. I want the app to auto update the contents of the AppPackage folder automatically on start. [Assume I have checked for updates and downloaded the required files to the data directory]

**Comment by rmarinho** [comment]
Theres no away for doing that on MAUI itself, but i moved to discussion so if other people have opinions about it.

**Comment by vickdini** [comment]
Hi guys! So basically what you want is a CI/CD pipeline for phone and Windows apps. It's pretty interesting if you think about it. Since you're looking for a way to update the apk without having to go through the store nor connect via cable to the development computer, you might want to look into using cloud storage, such as OneDrive or Mega, or if you'd rather use a more business-oriented approach: Azure, AWS, or GCP solutions. I know the first group of cloud storage I mentioned has phone apps that sync shared files.

So you could try creating a shared folder, setting up your project in VS to place the executable file there, and installing the cloud app on your phone and syncing the executable. The OS will probably ask for your authorization to install it, but this way you'd skip the cables and app stores.

It could be a great project to create the equivalent of docker, but for phone apps, thereby porting the MAUI hot reload feature to phones without using cables or even having VS installed, and testers would have access to the shared app in less time.

I hope this is helpful.
